### PR TITLE
Batch 5: Bahama

### PR DESCRIPTION
**Pull request by ericsorenson**
- 8653 [Court: Re-implement Notice Setting Case for Trial for In-Person Proceedings](https://github.com/flexion/ef-cms/issues/8653) 
- 8610 [Order Search: OR keyword search](https://github.com/flexion/ef-cms/issues/8610) 
- 8551 [Order Search: Display total results count](https://github.com/flexion/ef-cms/issues/8551) 
- 8550 [Order Search: Enhance document link behavior](https://github.com/flexion/ef-cms/issues/8550) 
- 8547 [Order Search: AND keyword search](https://github.com/flexion/ef-cms/issues/8547) 
- 8546 [Order Search: Locate document by exact keyword/phrase](https://github.com/flexion/ef-cms/issues/8546) 
- 8512 [Simplify the current Process for Migrations](https://github.com/flexion/ef-cms/issues/8512) 
- 8506 [Automate triggering of Blue/Green Migration](https://github.com/flexion/ef-cms/issues/8506)

### CODE REVIEW

File: web-api/migration-terraform/main/lambdas/migration-segments.js [hunk 1]
```diff
@@ -128,7 +160,14 @@ exports.handler = async event => {
     `about to process ${segment} of ${totalSegments}`,
   );
 
-  await scanTableSegment(segment, totalSegments);
+  const ranMigrations = {
+    ...(await hasMigrationRan('bug-0035-private-practitioner-representing.js')),
+    ...(await hasMigrationRan('bug-0036-public-served-parties-code.js')),
+    ...(await hasMigrationRan('0036-phone-number-format.js')),
```
Comment (adunkman) [inline web-api/migration-terraform/main/lambdas/migration-segments.js:166]: Is it problematic to have two migrations with code `0036`?

File: shared/src/business/useCases/users/createUserInteractor.test.js [hunk 2]
```diff
@@ -160,7 +160,42 @@ describe('create user', () => {
     });
   });
 
-  it('creates a legacyJudge user', async () => {
+  it('should create a generic user nd delete the barNumber when it is defined and the user is not a pracititoner', async () => {
```
Comment (adunkman) [inline shared/src/business/useCases/users/createUserInteractor.test.js]:
```suggestion
  it('should create a generic user and delete the barNumber when it is defined and the user is not a pracititoner', async () => {
```

File: web-api/elasticsearch/efcms-docket-entry-mappings.js [hunk 3]
```diff
@@ -26,10 +26,22 @@ module.exports = {
...
       type: 'text',
     },
     'documentTitle.S': {
       analyzer: 'ustc_analyzer',
+      fields: {
+        exact: {
+          analyzer: 'english_exact',
+          type: 'text',
+        },
+      },
```
Comment (adunkman) [inline web-api/elasticsearch/efcms-docket-entry-mappings.js:44]: Anything special needed to deploy these mapping changes?

File: web-api/elasticsearch/elasticsearch-settings.js [hunk 4]
```diff
@@ -12,6 +12,10 @@ module.exports = {
     index: {
       analysis: {
         analyzer: {
+          english_exact: {
+            filter: ['lowercase'],
+            tokenizer: 'standard',
+          },
```
Comment (adunkman) [inline web-api/elasticsearch/elasticsearch-settings.js:18]: Anything special needed to deploy the new settings?

**Review by adunkman — COMMENTED** [review]
Discovered in review, this PR also contains: 

- Devex 966
    - https://github.com/flexion/ef-cms/pull/8639
    - https://github.com/flexion/ef-cms/pull/8668
- Part of Devex 971 (no worries; can be accomplished in chunks)
    - https://github.com/flexion/ef-cms/pull/8644

⚠️ This pull request contains migrations which will need to be run on deploy, and an updated process to run migrations which will take effect _before_ the migrations are run for this pull request. To trigger migrations for this pull request (and going forward), follow the [process in the updated documentation](https://github.com/ustaxcourt/ef-cms/blob/a1e33bbf90b39bef055cd74e3626d83157c9fc57/docs/blue-green-migration.md).

A few quick questions dropped inline below!

File: web-api/elasticsearch/elasticsearch-settings.js (hunk 4 above)
Comment (ericsorenson) [reply web-api/elasticsearch/elasticsearch-settings.js:18]: It'll need an ElasticSearch re-index, which is required by the migrations.

File: web-api/elasticsearch/efcms-docket-entry-mappings.js (hunk 3 above)
Comment (ericsorenson) [reply web-api/elasticsearch/efcms-docket-entry-mappings.js:44]: It'll need an ElasticSearch re-index, which is required by the migrations.

File: web-api/migration-terraform/main/lambdas/migration-segments.js (hunk 1 above)
Comment (ericsorenson) [reply web-api/migration-terraform/main/lambdas/migration-segments.js:166]: They have unique names, so they should be fine.

File: shared/src/business/useCases/users/createUserInteractor.test.js (hunk 2 above)
Comment (ericsorenson) [reply shared/src/business/useCases/users/createUserInteractor.test.js]: Nice catch!

**Review by adunkman — APPROVED** [review]
Animated gif with the joke, "Where did the sheep go on vacation? The Bahaaaahmas"

**Pull request merged**